### PR TITLE
some cookie handling changes

### PR DESCRIPTION
**Pull request by thecristen**
Meant to address issues #46 and #41.
Also adds tests to verify that clicking "Check in another person" from the thank you page clears cookies.

### CODE REVIEW

**Comment by benpatterson** [comment]
Will a new cookie be stored when the form is submitted (i.e., will the initial user now see the form she filled out for her friend the next time she visits the site?) That could be a potential additional testcase if that's a concern.

**Comment by thecristen** [comment]
Yep! The successful form responses are [still stored as before](https://github.com/greenstreetsinitiative/checkin2015/blob/master/survey%2Fviews.py#L66)

[I'd thrown in an assertion for that](https://github.com/greenstreetsinitiative/checkin2015/pull/51/files#diff-5c3d2d005d122d11b35cc35b7da5e142R31) but was mostly focused on testing the biggest change introduced by this PR - where clicking "Check in another person" from the checkin confirmation page redirects you to a fresh copy of the form.

**Comment by benpatterson** [comment]
Och, no I meant moreso on the _next_ request (i.e., back to Bob), but I suppose that would be a new session altogether so perhaps I'm chasing up the wrong tree. Thanks for explaining. :)

File: survey/views.py [hunk 1]
```diff
@@ -108,36 +108,41 @@ def add_checkin(request):
...
-            except Employer.DoesNotExist:
-                initial_commute['employer'] = ''
-        if 'team' in request.session:
-            teamid = request.session.get('team')
-            if teamid:
-                try:
-                    initial_commute['team'] = Team.objects.get(pk=teamid)
-                except Team.DoesNotExist:
-                    initial_commute['team'] = ''
+        if request.GET.get('referral'):
```
Comment (benpatterson) [inline survey/views.py]: you may want to put a comment above this line explaining that we flush out the session when we detect it is a subsequent submittal as a referral.

File: survey/tests.py [hunk 2]
```diff
@@ -1,6 +1,36 @@
-from django.test import TestCase
+from django.test import TestCase, Client
+from django.conf import settings
+from django.utils.importlib import import_module
 import models
+import datetime
```
Comment (benpatterson) [inline survey/tests.py:9]: You could probably condense these lines to be `from datetime.datetime import date`.

**Comment by benpatterson** [comment]
BTW other than a couple nits, that was my only feedback on this PR. Thx for letting me give it a pass. LGTM.

File: survey/tests.py (hunk 2 above)
Comment (thecristen) [inline survey/tests.py:9]: just tried it and got `ImportError: No module named datetime` 😞

**Comment by thecristen** [comment]
Made more changes. No longer flushing the session.

File: survey/tests.py (hunk 2 above)
Comment (pwnage101) [inline survey/tests.py:9]: Hm, there doesn't seem to be any reason to import the date module anyway---try removing line 6 below?

**Comment by pwnage101** [comment]
attached preview of thanks page:

2016-05-20-141724_887x301_scrot

It's unclear which link to open in an incognito window.  I'll suggest beside the "check in another person" link, rather than below it.

**Comment by pwnage101** [comment]
Crap, sorry I didn't restart the server and there must have been cached css.  It displays correctly now and it looks fine.  This PR looks good.

File: survey/tests.py (hunk 2 above)
Comment (thecristen) [inline survey/tests.py:9]: That worked 👍 . Since there seem to be no other issues, I'll rebase and merge this in next.

**Pull request merged**